realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -888,26 +896,43 @@ protected void onSuccess(AuthenticateResponse response) {
                             scheduleRefreshAccessToken(authServer, response.getAccessToken().expiresMs());
                         }
                     }
+                    refreshTokenNetworkRequest = null;
                 }
             }
 
             @Override
             protected void onError(AuthenticateResponse response) {
                 if (!isClosed && !Thread.currentThread().isInterrupted()) {
                     onGoingAccessTokenQuery.set(false);
-                    RealmLog.error("Unrecoverable error, while refreshing the access Token (" + response.getError().toString() + ") reschedule will not happen");
+                    RealmLog.debug("Session[%s]: Unrecoverable error, while refreshing the access Token. Reschedule will not happen. %s", configuration.getPath(), response.getError());
                 }
+                refreshTokenNetworkRequest = null;
+            }
+
+            @Override
+            protected String getName() {
+                return taskName;
             }
-        });
-        refreshTokenNetworkRequest = new RealmAsyncTaskImpl(task, SyncManager.NETWORK_POOL_EXECUTOR);
+        }, SyncManager.NETWORK_POOL_EXECUTOR);
+    }
+
+    void refreshConnection() {
```

Comment:
Shouldn't this be  synchronized on the SyncSession instance? Is it not reachable from user code through `SyncManager.refreshConnection`?

Reply:
Yup. Good catch